.github/workflows/ts-test.yml

```diff
@@ -0,0 +1,63 @@
+name: Typescript Test
+
+on:
+  push:
+    paths:
+      - typescript-demo/**
+  pull_request:
+    paths:
+      - 'typescript-demo/**'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the repository
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Set up Node.js environment
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "22"
+
+      # Install dependencies in the `typescript-demo` folder
+      - name: Install dependencies
+        run: npm install
+        working-directory: ./typescript-demo
+
+      # Build the application
+      - name: Build the application
+        run: npm run build
+        working-directory: ./typescript-demo
+
+      # Start the application
+      - name: Start the application
+        run: npm start &
+        working-directory: ./typescript-demo
+        env:
+          NODE_ENV: test
+
+      # Wait for the application to be ready
+      - name: Wait for localhost:3000
+        uses: jakejarvis/wait-action@master
+        with:
+          url: "http://localhost:3000"
+          timeout: 60
+
+      # Run Jest tests
+      - name: Run Jest tests
+        run: npm test -- --coverage
+        working-directory: ./typescript-demo
+
```

Comment:
Have you tried to add  nexts actions from marketplace to see tests report and coverage directly in the Github action summary (as it is for Cypress) ? 
- [jest-test-report](https://github.com/marketplace/actions/jest-test-report)
- [jest-coverage-report](https://github.com/marketplace/actions/jest-coverage-report)

Reply:
No I didn't try, I'll try now